### DISCUSSION TITLE
Suricata Failure

### DISCUSSION THREAD

**Discussion by WPTechnician** [Security-Onion-Solutions/securityonion#2115] [Unsupported Versions] [answered]
Hi:

I am seeing that Suricata isn't working as expected.  I ran a so-status on the manager node and can see that the suricata service is in an error state.  The sensor node is reporting no errors.  The Suricata log on the manager/search node is reporting:

26/11/2020 -- 20:26:00 - <Notice> - This is Suricata version 5.0.4 RELEASE running in SYSTEM mode
26/11/2020 -- 20:26:00 - <Info> - CPUs/cores online: 32
26/11/2020 -- 20:26:00 - <Info> - dropped the caps for main thread
26/11/2020 -- 20:26:00 - <Info> - eve-log output device (regular) initialized: /nsm/eve-%Y-%m-%d-%H:%M.json
26/11/2020 -- 20:26:00 - <Info> - stats output device (regular) initialized: stats.log
26/11/2020 -- 20:26:00 - <Info> - Running in live mode, activating unix socket
26/11/2020 -- 20:26:02 - <Info> - 1 rule files processed. 20966 rules successfully loaded, 0 rules failed
26/11/2020 -- 20:26:02 - <Info> - Threshold config parsed: 32 rule(s) found
26/11/2020 -- 20:26:02 - <Info> - 20969 signatures processed. 1505 are IP-only rules, 3885 are inspecting packet payload, 15521 inspect application layer, 0 are decoder event only
26/11/2020 -- 20:26:10 - <Info> - can't get cpu-affinity node
26/11/2020 -- 20:26:10 - <Error> - [ERRCODE: SC_ERR_AFP_CREATE(190)] - Unable to find type for iface "bond0": No such device
26/11/2020 -- 20:26:10 - <Info> - Going to use 32 thread(s)
26/11/2020 -- 20:26:11 - <Info> - Running in live mode, activating unix socket
26/11/2020 -- 20:26:11 - <Info> - Using unix socket file '/var/run/suricata/suricata-command.socket'
26/11/2020 -- 20:26:11 - <Notice> - all 32 packet processing threads, 4 management threads initialized, engine started.
26/11/2020 -- 20:26:11 - <Error> - [ERRCODE: SC_ERR_AFP_CREATE(190)] - Unable to find iface bond0: No such device
26/11/2020 -- 20:26:11 - <Error> - [ERRCODE: SC_ERR_AFP_CREATE(190)] - Couldn't init AF_PACKET socket, fatal error
26/11/2020 -- 20:26:11 - <Error> - [ERRCODE: SC_ERR_FATAL(171)] - thread W#01-bond0 failed

Did I misconfigure something?  That interface doesn't exist on the manager/search node so it makes sense that it can't find it.  Thanks!

**Comment by WPTechnician** [comment]
Interesting, I am seeing new Alerts but barely any and not expected ones.  I tested something that I know should Alert (logged in to an unencrypted printer admin page) and although Zeek saw the traffic, Suricata did not.

**Reply by kiaraho** [reply]
Experiencing the same issue on a Manager after reconfiguring a Sensor (forward node) on the same grid. The sensor was reinstalled using the `securityonion-2.3.10.iso` file. No changes were made to the Manager. Contents of the `/opt/so/log/suricata/suricata.log` file from the Manager shown below, even though it does not have a monitoring interface.

```
26/11/2020 -- 04:45:21 - <Notice> - This is Suricata version 5.0.4 RELEASE running in SYSTEM mode
26/11/2020 -- 04:45:21 - <Info> - CPUs/cores online: 4
26/11/2020 -- 04:45:21 - <Info> - dropped the caps for main thread
26/11/2020 -- 04:45:21 - <Info> - eve-log output device (regular) initialized: /nsm/eve-%Y-%m-%d-%H:%M.json
26/11/2020 -- 04:45:21 - <Info> - stats output device (regular) initialized: stats.log
26/11/2020 -- 04:45:21 - <Info> - Running in live mode, activating unix socket
26/11/2020 -- 04:45:22 - <Info> - 1 rule files processed. 20983 rules successfully loaded, 0 rules failed
26/11/2020 -- 04:45:22 - <Info> - Threshold config parsed: 0 rule(s) found
26/11/2020 -- 04:45:22 - <Info> - 20986 signatures processed. 1507 are IP-only rules, 3898 are inspecting packet payload, 15525 inspect application layer, 0 are decoder event only
26/11/2020 -- 04:45:28 - <Info> - can't get cpu-affinity node
26/11/2020 -- 04:45:28 - <Error> - [ERRCODE: SC_ERR_AFP_CREATE(190)] - Unable to find type for iface "bond0": No such device
26/11/2020 -- 04:45:28 - <Info> - Going to use 4 thread(s)
26/11/2020 -- 04:45:28 - <Info> - Running in live mode, activating unix socket
26/11/2020 -- 04:45:28 - <Info> - Using unix socket file '/var/run/suricata/suricata-command.socket'
26/11/2020 -- 04:45:28 - <Notice> - all 4 packet processing threads, 4 management threads initialized, engine started.
26/11/2020 -- 04:45:28 - <Error> - [ERRCODE: SC_ERR_AFP_CREATE(190)] - Unable to find iface bond0: No such device
26/11/2020 -- 04:45:28 - <Error> - [ERRCODE: SC_ERR_AFP_CREATE(190)] - Couldn't init AF_PACKET socket, fatal error
26/11/2020 -- 04:45:28 - <Error> - [ERRCODE: SC_ERR_FATAL(171)] - thread W#01-bond0 failed
30/11/2020 -- 17:42:56 - <Notice> - This is Suricata version 5.0.4 RELEASE running in SYSTEM mode
30/11/2020 -- 17:42:56 - <Info> - CPUs/cores online: 4
30/11/2020 -- 17:42:56 - <Info> - dropped the caps for main thread
30/11/2020 -- 17:42:56 - <Info> - eve-log output device (regular) initialized: /nsm/eve-%Y-%m-%d-%H:%M.json
30/11/2020 -- 17:42:56 - <Info> - stats output device (regular) initialized: stats.log
30/11/2020 -- 17:42:56 - <Info> - Running in live mode, activating unix socket
30/11/2020 -- 17:42:56 - <Info> - 1 rule files processed. 1 rules successfully loaded, 0 rules failed
30/11/2020 -- 17:42:56 - <Info> - Threshold config parsed: 0 rule(s) found
30/11/2020 -- 17:42:56 - <Info> - 1 signatures processed. 0 are IP-only rules, 1 are inspecting packet payload, 0 inspect application layer, 0 are decoder event only
30/11/2020 -- 17:42:56 - <Info> - can't get cpu-affinity node
30/11/2020 -- 17:42:56 - <Error> - [ERRCODE: SC_ERR_AFP_CREATE(190)] - Unable to find type for iface "bond0": No such device
30/11/2020 -- 17:42:56 - <Info> - Going to use 4 thread(s)
30/11/2020 -- 17:42:56 - <Info> - Running in live mode, activating unix socket
30/11/2020 -- 17:42:56 - <Info> - Using unix socket file '/var/run/suricata/suricata-command.socket'
30/11/2020 -- 17:42:56 - <Notice> - all 4 packet processing threads, 4 management threads initialized, engine started.
30/11/2020 -- 17:42:56 - <Error> - [ERRCODE: SC_ERR_AFP_CREATE(190)] - Unable to find iface bond0: No such device
30/11/2020 -- 17:42:56 - <Error> - [ERRCODE: SC_ERR_AFP_CREATE(190)] - Couldn't init AF_PACKET socket, fatal error
30/11/2020 -- 17:42:56 - <Error> - [ERRCODE: SC_ERR_FATAL(171)] - thread W#01-bond0 failed
30/11/2020 -- 18:54:25 - <Notice> - This is Suricata version 5.0.4 RELEASE running in SYSTEM mode
30/11/2020 -- 18:54:25 - <Info> - CPUs/cores online: 4
30/11/2020 -- 18:54:25 - <Info> - dropped the caps for main thread
30/11/2020 -- 18:54:25 - <Info> - eve-log output device (regular) initialized: /nsm/eve-%Y-%m-%d-%H:%M.json
30/11/2020 -- 18:54:25 - <Info> - stats output device (regular) initialized: stats.log
30/11/2020 -- 18:54:25 - <Info> - Running in live mode, activating unix socket
30/11/2020 -- 18:54:27 - <Info> - 1 rule files processed. 20899 rules successfully loaded, 0 rules failed
30/11/2020 -- 18:54:27 - <Info> - Threshold config parsed: 0 rule(s) found
30/11/2020 -- 18:54:27 - <Info> - 20902 signatures processed. 1536 are IP-only rules, 3809 are inspecting packet payload, 15501 inspect application layer, 0 are decoder event only
30/11/2020 -- 18:54:33 - <Info> - can't get cpu-affinity node
30/11/2020 -- 18:54:33 - <Error> - [ERRCODE: SC_ERR_AFP_CREATE(190)] - Unable to find type for iface "bond0": No such device
30/11/2020 -- 18:54:33 - <Info> - Going to use 4 thread(s)
30/11/2020 -- 18:54:33 - <Info> - Running in live mode, activating unix socket
30/11/2020 -- 18:54:33 - <Info> - Using unix socket file '/var/run/suricata/suricata-command.socket'
30/11/2020 -- 18:54:33 - <Notice> - all 4 packet processing threads, 4 management threads initialized, engine started.
30/11/2020 -- 18:54:33 - <Error> - [ERRCODE: SC_ERR_AFP_CREATE(190)] - Unable to find iface bond0: No such device
30/11/2020 -- 18:54:33 - <Error> - [ERRCODE: SC_ERR_AFP_CREATE(190)] - Couldn't init AF_PACKET socket, fatal error
30/11/2020 -- 18:54:33 - <Error> - [ERRCODE: SC_ERR_FATAL(171)] - thread W#01-bond0 failed
```

**Reply by dougburks** [reply]
@kiaraho please create a separate discussion instead of adding to somebody else's discussion. Thanks!

**Reply by kiaraho** [reply]
@dougburks, I always search existing discussions before creating a new one, in case it is a duplicate issue. That being said, removing `so-suricata` from `/opt/so/conf/so-status/so-status.conf` like you suggested below, also worked for me.

**Reply by dougburks** [reply]
@kiaraho From the Support Guidelines (https://github.com/Security-Onion-Solutions/securityonion/discussions/1720):
> Start a new discussion instead of replying to an old one
Please search to see if you can find similar discussions that may help you. However, please do not reply to old discussions with your new issue. Instead, please start a new discussion and provide a hyperlink to the related discussion.

**Reply by kiaraho** [reply]
@dougburks, good to know. I wasn't sure if creating duplicate discussions would be an issue. Thanks!

**Comment by dougburks** [comment, marked as answer]
@WPTechnician You say `manager node` and then you say `manager/search node`. How exactly was the box configured? Typically, manager nodes and manager/search nodes do not run Suricata at all.

**Reply by WPTechnician** [reply]
Sorry, I wasn't clear, it is a manager/search node.  My configuration is that node with a sensor node.  This is the result of running so-status on the manager/search node:

![image](https://user-images.githubusercontent.com/19175877/100748965-066e0180-33b2-11eb-9e49-8c2262667947.png)

The log is in my original post.  Thanks!

**Reply by dougburks** [reply]
It sounds like at some point maybe you tried to apply the suricata state on your manager/search node. You can try removing `so-suricata` from `/opt/so/conf/so-status/so-status.conf`.

**Reply by WPTechnician** [reply]
Lol, I am learning!  Thanks very much for that Doug, that indeed worked to remove that error.
